Penyelesaian Sistem Persamaan Linear pada Rangkaian Listrik Menggunakan Metode Gauss-Jordan dengan Pivoting

Dalam bidang fisika, khususnya pada analisis rangkaian listrik, sering muncul sistem persamaan linear yang melibatkan beberapa variable arus dan tegangan. Untuk menyelesaikan sistem tersebut secara sistematis, dapat digunakan metode eliminasi Gauss-Jordan.

Metode ini memungkinkan transformasi matriks menjadi bentuk identitas sehingga solusi dapat diperoleh secara langsung. Dalam metode ini dilengkapi dengan teknik pivoting guna meningkatkan kestabilan numerik.

Metode Gauss-Jordan adalah pengembangan dari eliminasi Gauss yang bertujuan mengubah matriks menjadi bentuk identitas.

Metode ini bekerja melalui:


*   Normalisasi baris pivot
*   Eliminasi semua elemen selain pivot

Untuk menghindari kesalahan numrik, digunkana pivoting, yaitu menukar baris agar elemen pivot bernilai maksimum.
Salah satu penerapan SPL dalam fisika adalah pada hukum kirchoff. Yang digunakan untuk menganalisis arus dan tegangan dalam rangkaian listrik.



Berikut sebuah masalah fisika yang mengandung Sistem Persamaan Linear 4 variabel

Diberikan sebuah sistem persamaan:

I1 + I2 + I3 + I4 = 10

2I1 - I2 + I3 - I4 = 5

I1+3I2 - I3 + 2I4 = 8

3I1 - I2 + 2I3 + I4 = 7


In [1]:
import numpy as np

def gauss_jordan_pivot(A, b):
    n = len(b)
    M = np.hstack((A, b.reshape(-1,1)))

    for k in range(n):
        # Pivoting
        max_row = np.argmax(abs(M[k:, k])) + k
        M[[k, max_row]] = M[[max_row, k]]

        # Normalisasi pivot
        pivot = M[k, k]
        M[k] = M[k] / pivot

        # Eliminasi
        for i in range(n):
            if i != k:
                faktor = M[i, k]
                M[i] = M[i] - faktor * M[k]

    return M[:, -1]

# Matriks
A = np.array([
    [1, 1, 1, 1],
    [2, -1, 1, -1],
    [1, 3, -1, 2],
    [3, -1, 2, 1]
], dtype=float)

b = np.array([10, 5, 8, 7], dtype=float)

# Hasil
solusi = gauss_jordan_pivot(A, b)
print("Solusi:", solusi)

Solusi: [ 1.4  5.6  5.4 -2.4]


Dari hasil perhitungan diperoleh nilai arus:

I1 = 1.4
I2 = 5.6
I3 = 5.4
I4 = -2.4

Nilai ini menunjukkan besar arus pada masing-masing cabang rangkaian.

Jika terdapat nilai negatif, hal ini menunjukkan arah arus sebenarnya berlawanan dengan asumsi awal.

Berdasarkan hasil perhitungan menggunakan metode Gauss-Jordan dengan pivoting, diperoleh solusi yang memenuhi seluruh sistem persamaan. Hal ini menunjukkan bahwa implementasi program telah berjalan dengan benar.
Metode eliminasi Gauss-Jordan dengan pivoting dapat digunakan untuk menyelesaikan sistem persamaan linear dalam masalah fisika, khususnya pada analisis rangkaian listrik.

Metode ini memberikan solusi yang akurat dan efisien, serta mengurangi kesalahan numerik melalui proses pivoting.